src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2742,188 @@ def _convert_thermo_fisher_to_odemis_metadata(metadata: str) -> dict:
         logging.warning(f"Failed to parse ThermoFisher metadata: {model.MD_DET_TYPE} - {e}")
 
     return md
+
+### Tescan Metadata parsing
+def _parse_tescan_metadata(metadata: bytes) -> str:
+    """
+    Parses TESCAN metadata bytes to str
+    :param metadata: metadata bytes (TESCAN bytes format)
+    :return: (str) metadata (TESCAN ini format)
+    """
+    # NOTE:
+    # METADATA FORMAT
+    # JPEG2000 metadata
+    #   4 bytes: length of the next chunk (n)
+    #   2 bytes: header of the next chunk (type)
+    #   n bytes: chunk data
+    # MAIN metadata
+    #   4 bytes: length of the next chunk (n)
+    #   2 bytes: header of the next chunk (type)
+    #   n - 3 bytes: valid chunk data
+    #   3 bytes: unsure?
+    # SEM/FIB metadata
+    #   4 bytes: length of the next chunk (n)
+    #   2 bytes: header of the next chunk (type)
+    #   n - 3 bytes: valid chunk data
+    #   3 bytes: unsure?
+
+    # constants
+    _LENGTH_BYTES = 4
+    _HEADER_BYTES = 2
+    _END_BYTES = 3
+
+    # valid metadata str
+    vmd = ""
+
+    # header type map
+    _header_map ={
+        2: "[MAIN]",
+        3: "[SEM]",
+        5: "[FIB]"
+    }
+
+    # skip jpeg2000 metadata
+    next_chunk_len = int.from_bytes(metadata[:4], byteorder="little", signed=False)
+    
+    # loop through the two chunks, extract the valid data
+    for i in range(2):
```

Comment:
Note: this works fine for our needs. However, if you want to be more generic by just continuously scanning until either the `len(metadata)` < 6, or if encountering the type 0. Skip the chunks if the type is not in the header_map.